### ISSUE TITLE
Documentation is outdated?

### ISSUE THREAD

**Issue by segfall** [webui-dev/deno-webui#10]
I tried running the Minimal Example and the Repository Examples but neither seems to work anymore.
The former gives the error:
```
error: Uncaught SyntaxError: The requested module 'https://deno.land/x/webui/mod.ts' does not provide an export named 'WebUI'
import { WebUI } from "https://deno.land/x/webui/mod.ts";
         ^
    at <anonymous> (file:///path/to/garden/app.js:1:10)
```

and the latter gives the error:
```
error: Uncaught (in promise) Error: Could not open library: Could not open library: dlopen(/Users/jim/.deno_webui/ffd9edc4-webui-2.dyn, 0x0005): tried: '/Users/jim/.deno_webui/ffd9edc4-webui-2.dyn' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/jim/.deno_webui/ffd9edc4-webui-2.dyn' (no such file), '/Users/jim/.deno_webui/ffd9edc4-webui-2.dyn' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```

My current environment is like so:
>deno 1.36.3 (release, aarch64-apple-darwin)
>v8 11.6.189.12
>typescript 5.1.6
> Macbook Pro M2 Venture 13.2.1

Has anyone else encountered this issue?

**Comment by hassandraga** [comment]
Related to [this issue](https://github.com/webui-dev/deno-webui/issues/7)

**Comment by hassandraga** [comment]
I just tested it on Windows right now:

```
git clone https://github.com/webui-dev/deno-webui.git
cd deno-webui\examples\hello_world
deno run --allow-all --unstable hello_world.ts
```

![image](https://github.com/webui-dev/deno-webui/assets/16948659/528e4ea2-c600-46a1-9ec0-addd0deb2697)

**Comment by hassandraga** [comment]
I will re-test it on macOS tomorrow.

**Comment by segfall** [comment]
Thanks, Hassan. I got around the first issue by following the issue you linked and pulling the repo.

However, the Mac issue seems to be because WebUI expects a different CPU architecture.

**Comment by hassandraga** [comment]
You are right, we need to add the ARM64 lib into the repo. I will do it tomorrow.

**Comment by jeff-hykin** [comment]
Just a friendly bump that most of the stuff in the readme still doesn't run

**Comment by rmnoon** [comment]
Yep, running into the same stuff. Have it working on Apple Silicon by checking out the tree and using the library directly but the deno.land stuff is out of date. Also FWIW on latest MacOS with Chrome I'm just getting the UI opening in the full browser itself instead of the raw frame.

**Comment by hassandraga** [comment]
WebUI does not support Safari app mode yet. Firefox or any Chromium-based browser should be installed first to use app mode. Otherwise, WebUI will try to open the UI in the default macOS browser window, which will be in most cases Safari (_Not app-mode_).

On the other hand, WebUI v2.4.0 has been released... We will update the Deno-WebUI wrapper soon.

**Comment by jeff-hykin** [comment]
> Firefox or any Chromium-based browser should be installed first to use app mode

Okay then you probably want to change the Readme.md cause it says

> Fully Independent (No need for any third-party runtimes)

If the system needs a browser installed, then I wouldn't really call this library "fully independent". A browser is a JavaScript/Wasm runtime.

**Comment by AlbertShown** [comment]
> A browser is a third party

On one side, yes. In another, no. All programs in all OS need hundreds or thousands of libs, drivers, runtime, etc. All those are built-in in the OS, so nobody talks about it because it's already installed. A web browser is one of them.

Conversely, because Deno-webui needs Chromium browsers and not Safari, it is a third party then, as you said.

I suggest adding in the readme that a Chromium-based browser is needed.

**Comment by jeff-hykin** [comment]
> All programs in all OS need hundreds or thousands of libs

Actually some don't use a single system library. Take a look at [Cosmopolitan Libc](https://github.com/jart/cosmopolitan) not only does it statically link everything (does not link to a single system library at runtime), not only does it not call any system executables or need any env vars, but it can also make a single binary that runs on all OS's, even Windows.

I use these kinds of hyper-agressively independent binaries and tools for making reliable bootstrapping steps.

> because Deno-webui needs Chromium browsers and not Safari

You do make a good point though. I agree there is no way to make a graphical interface without using a system library; the tool always has to talk to whatever desktop environment the user has installed.


However, linking to something like libc is different than linking to a browser. Mac/Linux/Windows are all guaranteed (as part of the spec) to have libc. For browsers, Mac is guaranteed to have safari and windows is guaranteed to have edge. But Linux isn't required to have Firefox, or have it in the same location, and it can be updated independent of the OS version.

If there was support for Safari, then I would agree and say "it only needs a 2nd party runtime".

Saying "it needs a second party browser runtime" is very different than "no 3rd party runtime" which implies the runtime is provided and is consistent. The 2nd party runtime tell me behavior will be slightly different for each OS, and that there may be reliability issues on Linux.

**Comment by jeff-hykin** [comment]
Also can confirm readme example is working 👌  So this can probably be closed
